lib/src/json_pointer.dart

```diff
@@ -103,6 +103,14 @@ abstract class JsonPointer {
   @override
   String toString();
 
+  /// Compares the pointers paths for equality
```

Comment:
Why do we need to redefine these in the interface?

Reply:
It doesn't need them,  I started out implementing both methods in the abstract class and didn't remember to remove them when I moved them to JsonPointerSegment.   

There is a toString override already there in the interface that could be removed, it was already there so I left it alone.

Reply:
I pushed a new commit cleaning them up.